### PR TITLE
Update new gems

### PR DESCRIPTION
**Pull request by LocalIdentity**
Voltaxic Burst
Rage Vortex
Spectral Helix
Battlemage's Cry - Partial Support
Earthbreaker - Partial Support
Absolution
Manabond

### CODE REVIEW

**Review by PJacek — CHANGES_REQUESTED** [review]
You might want to check indentation. There is a few places where you used spaces instead of tabs

File: src/Data/Skills/act_dex.lua [hunk 1]
```diff
@@ -3349,11 +3349,52 @@ skills["ExplosiveConcoction"] = {
...
+		{
+			name = "All Flasks",
+		},
+	},
+	statMap = {
+		["flask_throw_minimum_cold_damage_if_used_sapphire_flask"] = {
+			mod("ColdMin", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8 } }),
+		},
+		["flask_throw_maximum_cold_damage_if_used_sapphire_flask"] = {
+			mod("ColdMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8} }),
```
Comment (ppoelzl) [inline src/Data/Skills/act_dex.lua:3383]:
```suggestion
			mod("ColdMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8 } }),
```

File: src/Data/Skills/act_dex.lua [hunk 2]
```diff
@@ -3349,11 +3349,52 @@ skills["ExplosiveConcoction"] = {
...
+		},
+		["flask_throw_minimum_lightning_damage_if_used_topaz_flask"] = {
+			mod("LightningMin", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8 } }),
+		},
+		["flask_throw_maximum_lightning_damage_if_used_topaz_flask"] = {
+			mod("LightningMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8} }),
```
Comment (ppoelzl) [inline src/Data/Skills/act_dex.lua:3389]:
```suggestion
			mod("LightningMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8 } }),
```

File: src/Data/Skills/act_int.lua [hunk 3]
```diff
@@ -9165,9 +9167,22 @@ skills["SummonReaper"] = {
...
 	castTime = 1,
+	minionList = {
+		"SummonedReaper",
+	},
+	statMap = {
+		["bleed_on_hit_with_attacks_%"] = {
+			mod("MinionModifier", "LIST", { mod = mod("BleedChance", "BASE", nil, ModFlag.Attack) })
+		},
+		["non_reaper_minion_damage_+%_final"] = {
+			mod("MinionModifier", "LIST", { mod = mod("Damage", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true  }),
```
Comment (ppoelzl) [inline src/Data/Skills/act_int.lua:9178]:
```suggestion
			mod("MinionModifier", "LIST", { mod = mod("Damage", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true }),
```

File: src/Data/Skills/act_int.lua [hunk 4]
```diff
@@ -9165,9 +9167,22 @@ skills["SummonReaper"] = {
...
+		},
+		["non_reaper_minion_maximum_life_+%_final"] = {
+			mod("MinionModifier", "LIST", { mod = mod("Life", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true  }),
```
Comment (ppoelzl) [inline src/Data/Skills/act_int.lua:9181]:
```suggestion
			mod("MinionModifier", "LIST", { mod = mod("Life", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true }),
```

File: src/Data/Skills/sup_str.lua [hunk 5]
```diff
@@ -259,6 +259,11 @@ skills["SupportBehead"] = {
 	addSkillTypes = { },
 	excludeSkillTypes = { },
 	statDescriptionScope = "gem_stat_descriptions",
+	statMap = {
+		["support_executioner_damage_vs_enemies_on_low_life_+%_final"] = {
+			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "ActorCondition", actor = "enemy", var = "LowLife"})
```
Comment (ppoelzl) [inline src/Data/Skills/sup_str.lua:264]:
```suggestion
			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "ActorCondition", actor = "enemy", var = "LowLife" })
```

File: src/Export/Skills/act_dex.txt [hunk 6]
```diff
@@ -611,7 +611,50 @@ local skills, mod, flag, skill = ...
...
+		{
+			name = "All Flasks",
+		},
+	},
+	statMap = {
+		["flask_throw_minimum_cold_damage_if_used_sapphire_flask"] = {
+			mod("ColdMin", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8 } }),
+		},
+		["flask_throw_maximum_cold_damage_if_used_sapphire_flask"] = {
+			mod("ColdMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8} }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_dex.txt:646]:
```suggestion
			mod("ColdMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 2, 5, 6, 8 } }),
```

File: src/Export/Skills/act_dex.txt [hunk 7]
```diff
@@ -611,7 +611,50 @@ local skills, mod, flag, skill = ...
...
+		},
+		["flask_throw_minimum_lightning_damage_if_used_topaz_flask"] = {
+			mod("LightningMin", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8 } }),
+		},
+		["flask_throw_maximum_lightning_damage_if_used_topaz_flask"] = {
+			mod("LightningMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8} }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_dex.txt:652]:
```suggestion
			mod("LightningMax", "BASE", nil, 0, 0, { type = "SkillPart", skillPartList = { 3, 5, 7, 8 } }),
```

File: src/Export/Skills/act_dex.txt [hunk 8]
```diff
@@ -611,7 +611,50 @@ local skills, mod, flag, skill = ...
...
+		},
+		["flask_throw_ruby_flask_ignite_damage_+%_final"] = {
+			mod("Damage", "MORE", nil, 0, KeywordFlag.Ignite, { type = "SkillPart", skillPartList = { 4, 6, 7, 8} }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_dex.txt:655]:
```suggestion
			mod("Damage", "MORE", nil, 0, KeywordFlag.Ignite, { type = "SkillPart", skillPartList = { 4, 6, 7, 8 } }),
```

File: src/Export/Skills/act_int.txt [hunk 9]
```diff
@@ -555,12 +563,12 @@ local skills, mod, flag, skill = ...
...
 	},
 	statMap = {
-	    ["firestorm_initial_impact_damage_+%_final"] = {
-	        mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 } )
-	    },
-	    ["firestorm_initial_impact_area_of_effect_+%_final"] = {
-	        mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 } )
-	    },
+		["firestorm_initial_impact_damage_+%_final"] = {
+			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 } )
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:567]:
```suggestion
			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 })
```

File: src/Export/Skills/act_int.txt [hunk 10]
```diff
@@ -555,12 +563,12 @@ local skills, mod, flag, skill = ...
...
+		},
+		["firestorm_initial_impact_area_of_effect_+%_final"] = {
+			mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 } )
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:570]:
```suggestion
			mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 })
```

File: src/Data/Skills/act_int.lua [hunk 11]
```diff
@@ -3208,12 +3211,12 @@ skills["Firestorm"] = {
...
 	},
 	statMap = {
-	    ["firestorm_initial_impact_damage_+%_final"] = {
-	        mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 } )
-	    },
-	    ["firestorm_initial_impact_area_of_effect_+%_final"] = {
-	        mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 } )
-	    },
+		["firestorm_initial_impact_damage_+%_final"] = {
+			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 } )
```
Comment (ppoelzl) [inline src/Data/Skills/act_int.lua:3215]:
```suggestion
			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "SkillPart", skillPart = 1 })
```

File: src/Data/Skills/act_int.lua [hunk 12]
```diff
@@ -3208,12 +3211,12 @@ skills["Firestorm"] = {
...
+		},
+		["firestorm_initial_impact_area_of_effect_+%_final"] = {
+			mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 } )
```
Comment (ppoelzl) [inline src/Data/Skills/act_int.lua:3218]:
```suggestion
			mod("AreaOfEffect", "MORE", nil, 0, 0, { type = "SkillPart", skillPart = 1 })
```

File: src/Export/Skills/act_int.txt [hunk 13]
```diff
@@ -1166,7 +1174,15 @@ local skills, mod, flag, skill = ...
 #mods
 
 #skill Manabond
-#flags spell area lightning arcane
+#flags spell area arcane
+	statMap = {
+		["mana_void_gain_%_missing_unreserved_mana_as_base_lightning_damage"] = {
+			mod("Multiplier:ManaBondUnreservedMana", "BASE", nil, 0, 0, { type = "PerStat", stat = "ManaUnreserved" }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:1180]:
```suggestion
			mod("Multiplier:ManabondUnreservedMana", "BASE", nil, 0, 0, { type = "PerStat", stat = "ManaUnreserved" }),
```

File: src/Export/Skills/act_int.txt [hunk 14]
```diff
@@ -1166,7 +1174,15 @@ local skills, mod, flag, skill = ...
...
+			div = 100,
+		},
+	},
+#baseMod mod("LightningMin", "BASE", 1, 0, 0, { type = "Multiplier", var = "ManaBondUnreservedMana" })
+#baseMod mod("LightningMax", "BASE", 1, 0, 0, { type = "Multiplier", var = "ManaBondUnreservedMana" })
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:1185]:
```suggestion
#baseMod mod("LightningMin", "BASE", 1, 0, 0, { type = "Multiplier", var = "ManabondUnreservedMana" })
#baseMod mod("LightningMax", "BASE", 1, 0, 0, { type = "Multiplier", var = "ManabondUnreservedMana" })
```

File: src/Export/Skills/act_int.txt [hunk 15]
```diff
@@ -1618,7 +1634,21 @@ local skills, mod, flag, skill = ...
...
+#flags spell minion
+	minionList = {
+		"SummonedReaper",
+	},
+	statMap = {
+		["bleed_on_hit_with_attacks_%"] = {
+			mod("MinionModifier", "LIST", { mod = mod("BleedChance", "BASE", nil, ModFlag.Attack) })
+		},
+		["non_reaper_minion_damage_+%_final"] = {
+			mod("MinionModifier", "LIST", { mod = mod("Damage", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true  }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:1646]:
```suggestion
			mod("MinionModifier", "LIST", { mod = mod("Damage", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true }),
```

File: src/Export/Skills/act_int.txt [hunk 16]
```diff
@@ -1618,7 +1634,21 @@ local skills, mod, flag, skill = ...
...
+		},
+		["non_reaper_minion_maximum_life_+%_final"] = {
+			mod("MinionModifier", "LIST", { mod = mod("Life", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true  }),
```
Comment (ppoelzl) [inline src/Export/Skills/act_int.txt:1649]:
```suggestion
			mod("MinionModifier", "LIST", { mod = mod("Life", "MORE", nil) }, 0, 0, { type = "SkillName", skillName = "Summon Reaper", neg = true }, { type = "GlobalEffect", effectType = "Buff", unscalable = true }),
```

File: src/Export/Skills/sup_str.txt [hunk 17]
```diff
@@ -33,14 +33,18 @@ local skills, mod, flag, skill = ...
 #mods
 
 #skill SupportBehead
-#flags strike
+	statMap = {
+		["support_executioner_damage_vs_enemies_on_low_life_+%_final"] = {
+			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "ActorCondition", actor = "enemy", var = "LowLife"})
```
Comment (ppoelzl) [inline src/Export/Skills/sup_str.txt:38]:
```suggestion
			mod("Damage", "MORE", nil, 0, bit.bor(KeywordFlag.Hit, KeywordFlag.Ailment), { type = "ActorCondition", actor = "enemy", var = "LowLife" })
```

File: src/Modules/CalcPerform.lua [hunk 18]
```diff
@@ -863,6 +863,7 @@ local function doActorMisc(env, actor)
 		end
 		if modDB:Flag(nil, "Condition:CanGainRage") or modDB:Sum("BASE", nil, "RageRegen") > 0 then
 			output.MaximumRage = modDB:Sum("BASE", skillCfg, "MaximumRage")
+			modDB.multipliers["MaxRageVortexSacrifice"] = output.MaximumRage / 4
```
Comment (ppoelzl) [inline src/Modules/CalcPerform.lua:866]: Poedb says that you can sacrifice 20% of rage, not 25%: https://poedb.tw/us/Rage_Vortex Also, shouldn't this just be a normal multiplier mod as well, or is this not possible due to the evaluation order?

File: src/Modules/CalcPerform.lua [hunk 19]
```diff
@@ -1143,6 +1144,20 @@ function calcs.perform(env, avoidCache)
 					env.player.modDB:NewMod("CoveredInAshEffect", "BASE", infernalAshEffect * uptime, { type = "Multiplier", var = "WarcryPower", div = 5 })
 				end
 				modDB:NewMod("InfernalActive", "FLAG", true) -- Prevents effect from applying multiple times
+			elseif activeSkill.activeEffect.grantedEffect.name == "Battlemage's Cry" and not modDB:Flag(nil, "BattlemageActive") then
+				local battlemageSpellToAttack = activeSkill.skillModList:Sum("BASE", env.player.mainSkill.skillCfg, "BattlemageSpellIncreaseApplyToAttackPer5MP")
+				local battlemageSpellToAttackMax = m_floor(150 * buff_inc)
```
Comment (ppoelzl) [inline src/Modules/CalcPerform.lua:1149]: Consider refactoring this magic number to `data.misc`.

File: src/Modules/CalcPerform.lua [hunk 20]
```diff
@@ -1143,6 +1144,20 @@ function calcs.perform(env, avoidCache)
...
+				local battlemageCritChance = activeSkill.skillModList:Sum("BASE", env.player.mainSkill.skillCfg, "BattlemageCritChancePer5MP")
+				local battlemageCritChanceMax = m_floor(30 * buff_inc)
```
Comment (ppoelzl) [inline src/Modules/CalcPerform.lua:1151]: Consider refactoring this magic number to `data.misc`.

**Review by ppoelzl — CHANGES_REQUESTED** [review]

File: src/Modules/CalcPerform.lua (hunk 18 above)
Comment (ppoelzl) [reply src/Modules/CalcPerform.lua:866]: Update: This is support for helmet enchantment. Thanks, LI.

**Comment by livejamie** [comment]
Thanks for all your hard work guys! Looking forward to Forbidden Rite. :)

**Pull request merged**